test/e2e/nodenetworkpolicy_test.go

```diff
@@ -39,22 +39,40 @@ func initializeAntreaNodeNetworkPolicy(t *testing.T, data *TestData, toHostNetwo
 	podsPerNamespace = []string{"a"}
 	suffix := randName("")
 	namespaces = make(map[string]TestNamespaceMeta)
-	for _, ns := range []string{"x", "y", "z"} {
-		namespaces[ns] = TestNamespaceMeta{
-			Name: ns + "-" + suffix,
-		}
-	}
 	nodes = make(map[string]string)
-	nodes["x"] = controlPlaneNodeName()
-	nodes["y"] = workerNodeName(1)
 	hostNetworks := make(map[string]bool)
+
+	// Deploy a hostNetwork Pod on a Node.
+	nodes["x"] = controlPlaneNodeName()
 	hostNetworks["x"] = true
-	if toHostNetworkPod {
+	namespaces["x"] = TestNamespaceMeta{
+		Name: "x-" + suffix,
+	}
+
+	if testNodeToNode {
+		// To test NodeNetworkPolicy between Nodes, deploy another hostNetwork Pod on another Node.
+		nodes["y"] = workerNodeName(1)
 		hostNetworks["y"] = true
+		namespaces["y"] = TestNamespaceMeta{
+			Name: "y-" + suffix,
+		}
 	} else {
+		// To test NodeNetworkPolicy between Nodes and Pods, create another two non-hostNetwork Pods on another Node.
+		// Ideally, a non-hostNetwork Pod should be deployed on the same Node where the hostNetwork Pod is located.
```

Comment:
> Ideally, a non-hostNetwork Pod should be deployed on the same Node where the hostNetwork Pod is located.

That's not what the code is doing though. The "hostNetwork Pod" is created with `controlPlaneNodeName()`, and these 2 Pods are created with `workerNodeName(1)`.

Reply:
I have polished this part. Hopefully, it will be easier to understand.